WebPhapp/backend/server.js

```diff
@@ -589,6 +601,129 @@ app.get('/api/v1/patients/:patientID', (req,res) => {
     res.json(matchingPatient);
 });
 
+/////////////
+/// users //
+////////////
+
+/*
+About:
+    The api endpoint to create a user.
+    Expects a username, password and role.
+Examples:
+    curl 'http://localhost:5000/api/v1/users/add' -H 'Acceptapplication/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"username":"mdulin2","password":"jacob","role":"Patient"}'
+Returns:
+    Success or failure with message
+*/
+app.post('/api/v1/users/add', (req,res) => {
+    const userInfo = req.body;
+
+    // validate fields exist that should
+    fields = [
+        userInfo.username,
+        userInfo.password,
+        userInfo.role
+    ];
+    fieldsSet = new Set(fields);
+    if(fieldsSet.has(undefined) || fieldsSet.has(null)){
+        // How to return a fail here?
+        console.log("/api/v1/users/add: error: Not all fields");
+        res.status(400).send(false);
+        return;
+    }
+
+    // Salts: Making the random table attack near impossible.
+    var salt = crypto.randomBytes(64).toString('base64');
+
+    // Hash the password
+    var hashedPassword = pbkdf2.pbkdf2Sync(userInfo.password, salt, 1, 32, 'sha512').toString('base64');
```

Comment:
I don't know much, but that seems like a beefy SHA. Nice.

Reply:
Just curious, did you consider the asnyc version of this function?

Reply:
PBKDF2 are key derivation functions that happen to be slower because of this fact. Yes, SHA512 is a good hashing algorithm. But, by itself, it is much, much too fast. 

I do not like dealing with promises and such. So, when I saw the sync version I went with that. Any problems?